utils/build.ps1

```diff
@@ -571,6 +571,8 @@ if (-not $PinnedBuild) {
 }
 
 $PinnedToolchain = [IO.Path]::GetFileNameWithoutExtension($PinnedBuild)
+# Use a shorter name in paths to avoid going over the path length limit.
+$NormalizedToolchainName = $PinnedToolchain -replace 'swift-(.+?)-a-windows10.*', '$1'
```

Comment:
I'm not sure that `NormalizedToolchainName` really is helpful, especially at a global level. It doesn't tell you why it is useful or what the normalisation is. This really is the version toolchain identifier.

The regex doesn't entirely work though, consider the following reasonable value: swift-6.2.1-RELEASE-windows10.exe. This should give us `6.2.1` (or perhaps `6.2.1-RELEASE` if we are trying to match the fact that we retain the `DEVELOPMENT-SNAPSHOT` for the development snapshots, but the versioning there would result in collisions otherwise I think).